### PR TITLE
Documentation for database stubbing using Specmatic

### PR DESCRIPTION
**Pull request by deshmukhsd**
Documentation for database stubbing using Specmatic.

### CODE REVIEW

File: documentation/stubbing_database.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
+---
+layout: default
+title: Troubleshooting
+parent: Documentation
+nav_exclude: true
+---
+
+## Introduction to Database Stubbing
+Database is an organized collection of structured information, or data, typically stored electronically in a computer system, so that it can be easily accessed and managed.
```
Comment (joelrosario) [inline documentation/stubbing_database.md]: We can remove this line.

File: documentation/stubbing_database.md [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+
+This documentation describes how to stub out Database using JUnit 5
+
+### Pre-requisite Setup
+Add Below-mentioned dependencies in pom.xml .Use the latest version of the dependencies.
```
Comment (joelrosario) [inline documentation/stubbing_database.md]: Fix the full-stop spacing

File: documentation/stubbing_database.md [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+   <version>0.61.6-db</version>
+   <exclusions>
+       <exclusion>
+           <groupId>in.specmatic</groupId>
+           <artifactId>specmatic-core</artifactId>
+       </exclusion>
+   </exclusions>
+</dependency>
+
+<dependency>
```
Comment (joelrosario) [inline documentation/stubbing_database.md:29]: This was a project-specific dependency I believe, we may need to separate this out and say that sometimes there is a Xerces library version conflict, and in this case you could find out the version of Xerces used by Specmatic and pin it. 

Then give this dependency as an example.

File: documentation/stubbing_database.md [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+public JdbcMockFactory jdbcMockFactory() {
+    if (jdbcMockFactory == null) jdbcMockFactory = new JdbcMockFactory(new ExternalStub("localhost", 9000));
+    if (mockDataSource == null) mockDataSource = jdbcMockFactory.createDataSource();
+    return jdbcMockFactory;
+}
+```
+ - From above code you can return DataSource object created in jdbcMockFactory method to your DataSource.
+ - Add below property in respective profile application properties file.
+
+        spring.main.allow-bean-definition-overriding=true
```
Comment (joelrosario) [inline documentation/stubbing_database.md]: Should be quoted using triple tilde

**Review by joelrosario — CHANGES_REQUESTED** [review]

File: documentation/stubbing_database.md (hunk 1 above)
Comment (deshmukhsd) [reply documentation/stubbing_database.md]: removed

File: documentation/stubbing_database.md (hunk 2 above)
Comment (deshmukhsd) [reply documentation/stubbing_database.md]: fixed the spacing

File: documentation/stubbing_database.md (hunk 3 above)
Comment (deshmukhsd) [reply documentation/stubbing_database.md:29]: Added explanation for dependency

File: documentation/stubbing_database.md (hunk 4 above)
Comment (deshmukhsd) [reply documentation/stubbing_database.md]: Added quotation with tilde

File: documentation/stubbing_database.md [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+
+<dependency>
+   <groupId>in.specmatic</groupId>
+   <artifactId>specmatic-core</artifactId>
+   <version>0.65.1</version>
+   <scope>test</scope>
+</dependency>
+```
+
+- Sometimes there is a Xerces library version conflict, and in this case, you could find out the version of Xerces used by Specmatic.
```
Comment (joelrosario) [inline documentation/stubbing_database.md]: "... the version of Xerces used by Specmatic, and pin it in the pom.xml dependencies."

File: documentation/stubbing_database.md [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+   <groupId>xerces</groupId>
+   <artifactId>xercesImpl</artifactId>
+   <version>2.12.0</version>
+   <scope>test</scope>
+</dependency>
+```
+
+### Setup guidelines
+
+There are two approaches to stub out the database as follows:
```
Comment (joelrosario) [inline documentation/stubbing_database.md]: remove "as follows"

File: documentation/stubbing_database.md [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+parent: Documentation
+nav_exclude: true
+---
+
+## Introduction to Database Stubbing
+
+This documentation describes how to stub out the Database using JUnit 5.
+
+### Pre-requisite Setup
+Add the below-mentioned dependencies in `pom.xml`.Use the latest version of the dependencies.
```
Comment (joelrosario) [inline documentation/stubbing_database.md]: Space after the fullstops

**Review by joelrosario — CHANGES_REQUESTED** [review]

File: documentation/stubbing_database.md (hunk 7 above)
Comment (deshmukhsd) [reply documentation/stubbing_database.md]: added space

File: documentation/stubbing_database.md (hunk 5 above)
Comment (deshmukhsd) [reply documentation/stubbing_database.md]: added the explanation

File: documentation/stubbing_database.md (hunk 6 above)
Comment (deshmukhsd) [reply documentation/stubbing_database.md]: removed

**Review by joelrosario — APPROVED** [review]

**Pull request merged**